### ISSUE TITLE
Work on generating contract obj typescript interface dynamically using abi

### ISSUE THREAD

**Issue by zemse** [KMPARDS/esn-contracts#10]

**Comment by zemse** [comment]
Things to have a look: 
- [https://stackoverflow.com/questions/45771307/typescript-dynamically-create-interface](https://stackoverflow.com/questions/45771307/typescript-dynamically-create-interface)
- [https://stackoverflow.com/questions/53636756/typescript-interface-extending-another-interface-with-nested-properties](https://stackoverflow.com/questions/53636756/typescript-interface-extending-another-interface-with-nested-properties)
- [https://github.com/Soltsice/Soltsice](https://github.com/Soltsice/Soltsice) Looks like this is not compatible with ethers.js. The concept is nice though. Just need to add a command `soltsice ./build ./test/typings` when node compile.js is run which updates the interface files.

**Comment by zemse** [comment]
For now, create static interface files which would have to manually updated with contract methods are changed.

**Comment by zemse** [comment]
Just as an update, yesterday I forked [Typechain](https://github.com/ethereum-ts/TypeChain) project and implemented ethers v5 target into it. BTW, TypeChain takes in the contract artifact files and spits out type definition files. Just need to import them. However, it has ethers v4 by default and there are substantial changes in ethers v5 API. As I write more test cases, I'll identify more issues and I'll add more v5 features into the custom target package that I've made.
